src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can make a destructuring to get all the properties of action that you need to make your code cleaner

Reply:
not fixed
`const { type,  extraData, keysToRemove } = action`